src/app/containers/app-header/app-header.component.ts

```diff
@@ -108,7 +108,7 @@ export class AppHeaderComponent implements OnInit, OnDestroy {
 
   public async setHeaderContent(userDetails) {
     if (userDetails.userInfo) {
-      this.userRoles = userDetails.userInfo.roles;
+      this.userRoles = userDetails?.userInfo?.roles;
```

Comment:
Sorry to be a pain, but you're already inside if(userDetails.userInfo), so you don't need to use optional chaining on line 111 (though your might want to on line 110). Doesn't really matter I suppose